### PR TITLE
fix: Translate ISO 8601 Datetime Durations During Scraping/Parsing/Migrating

### LINKED ISSUE #2807: [SCRAPER] - Units for "Servings", "Minutes" and "Hour" missing  in German localization
### First Check

- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have verified that this issue _is not_ related to the underlying library
[hhyrsev/recipe-scrapers](https://github.com/hhursev/recipe-scrapers) by **1)** checking
the [debugger](https://demo.mealie.io/g/home/r/create/debug) and data is returned, **2)**
verifying that there _are_ errors in the log related to application level code, or
**3)** verified that the site provides recipe data, or is otherwise supported by
[hhyrsev/recipe-scrapers](https://github.com/hhursev/recipe-scrapers)

- [X] This issue can be replicated on the demo site (https://demo.mealie.io/)

### Please provide 1-5 example URLs that are having errors

https://www.chefkoch.de/rezepte/1313551236082544/Baumkuchen.html

### Please provide your logs for the Mealie container `docker logs <container-id> > mealie.logs`

[2023-12-09-mealie.log](https://github.com/mealie-recipes/mealie/files/13624577/2023-12-09-mealie.log)

### Deployment

Docker (Linux)

### PR DESCRIPTION
**Pull request by michael-genson**
## What type of PR is this?

_(REQUIRED)_

- bug

## What this PR does / why we need it:

_(REQUIRED)_

Adds backend translations to the ISO 8601 time duration parsing

## Which issue(s) this PR fixes:

_(REQUIRED)_

Mostly fixes https://github.com/mealie-recipes/mealie/issues/2807 (see below)

## Special notes for your reviewer:

_(fill-in or delete this section)_

Most of the PR is properly propagating the localizer down to the scraper/cleaning functions. I also had to implement some logic to properly parse "plural" translations. I'm not sure why we have a manual implementation for i18n parsing on the backend, but I basically [followed the Vue docs](https://kazupon.github.io/vue-i18n/guide/pluralization.html) and mirrored that functionality (and added tests)

Part of https://github.com/mealie-recipes/mealie/issues/2807 mentions that "serving" isn't translated, but looking at the example website source there is no word (it's just the number "1"). I tried my best to find where the word "serving" gets added, but I can't find it. I think it's somewhere deep in the recipe scrapers library, but I'm not sure. It's not in Mealie (I did a painfully exhaustive search for the word "serving" in the codebase). I thought it might get added in the cleaning process, but it definitely doesn't

## Testing

_(fill-in or delete this section)_

Pytest

### CODE REVIEW

**Comment by michael-genson** [comment]
It takes like 18 years for tests to run on my instance but it took 5 minutes to run on GH...

File: mealie/services/scraper/cleaner.py [hunk 1]
```diff
@@ -374,7 +375,7 @@ def clean_time(time_entry: str | timedelta | None) -> None | str:
             # TODO: Not sure what to do here
             return str(time_entry)
         case _:
-            logger.warning("[SCRAPER] Unexpected type or structure for time_entrys")
+            logger.warning("[SCRAPER] Unexpected type or structure for time_entries")
```
Comment (boc-the-git) [inline mealie/services/scraper/cleaner.py]: I accept that this is more correct English, but I actually don't like it. Reason being that I want the value in the message to be something I can search, and `time_entries` doesn't get me that. Nor does `time_entrys` to be fair.

I wonder if this is better as `[SCRAPER] Unexpected type or structure for variable time_entry`

File: tests/unit_tests/test_recipe_parser.py [hunk 2]
```diff
@@ -18,9 +19,10 @@
 @pytest.mark.parametrize("recipe_test_data", test_cases)
 @pytest.mark.asyncio
 async def test_recipe_parser(recipe_test_data: RecipeSiteTestCase):
-    recipe, _ = await scraper.create_from_url(recipe_test_data.url)
+    translator = local_provider()
+    recipe, _ = await scraper.create_from_url(recipe_test_data.url, translator)
 
     assert recipe.slug == recipe_test_data.expected_slug
-    assert len(recipe.recipe_instructions) == recipe_test_data.num_steps
+    assert len(recipe.recipe_instructions or []) == recipe_test_data.num_steps
```
Comment (boc-the-git) [inline tests/unit_tests/test_recipe_parser.py:26]: @michael-genson this is syntax I'm not familiar with.. is this ultimately saying "give me the length of `recipe.recipe_instructions` if it is defined/assigned/exists, or the length of the array `[]` (i.e. 0) if it isn't"?

Comment (michael-genson) [reply tests/unit_tests/test_recipe_parser.py:26]: Yup, that's correct. If `recipe.recipe_instructions` is `None`, `len(None)` will fail. Another more verbose way of writing it in one line:
```python
assert len(recipe.recipe_instructions) if recipe.recipe_instructions else 0 == recipe_test_data.num_steps
```

Or, more clearly in a few lines:
```python
if recipe.recipe_instructions:
    assert len(recipe.recipe_instructions) == recipe_test_data.num_steps
else:
    assert recipe_test_data.num_steps == 0  # or assert not recipe_test_data.num_steps
```

But the `or` syntax is super nice, and easy to read once you're familiar with it (it threw me for a loop the first time I saw it, too)

File: mealie/services/scraper/cleaner.py (hunk 1 above)
Comment (michael-genson) [reply mealie/services/scraper/cleaner.py]: I genuinely do not remember changing this, lol. I agree with your alternative

**Review by boc-the-git — APPROVED** [review]
LGTM!

**Pull request merged**